codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2UnknownFrame.java

```diff
@@ -119,18 +119,19 @@ public boolean equals(Object o) {
             return false;
         }
         DefaultHttp2UnknownFrame other = (DefaultHttp2UnknownFrame) o;
-        return super.equals(other) && flags().equals(other.flags())
-                && frameType() == other.frameType() && (stream() == null && other.stream() == null) ||
-                stream().equals(other.stream());
+        return (stream == other.stream() || stream != null && stream.equals(other.stream()))
```

Comment:
nit: could read `other.stream()` into a var

Reply:
I think will be better to use `java.util.Objects#equals`. But it is unsuitable for branch 4.1. If this is not a problem, I will change to `Objects#equals`.

Reply:
@fenik17 I was just suggesting to add a line before like
```java
Http2FrameStream otherStream = other.stream();
```

Reply:
@njhill yes I understood )
I suggest it would be better to use such approach:
```java
        return Objects.equals(stream, other.stream)
               && flags.equals(other.flags)
               && frameType == other.frameType
               && super.equals(other);
```
But this only for Java 8.

Reply:
@fenik17 I understood your suggestion too :) you're right that we can't do it for 4.1

Reply:
done